### ISSUE TITLE
Realm.schemaVersion returns 18446744073709552000 instead of -1 in test

### ISSUE THREAD

**Issue by Yupeng-li** [realm/realm-js#6119]
### How frequently does the bug occur?

Always

### Description

I use jest to test the realm related logic. Since Realm v12, some of tests start failing because Realm.schemaVersion doesn't return -1 if realm file doesn't exist.  Below is a quick test to show the issue.

```
expect(Realm.schemaVersion('./default.realm')).toBe(-1)

-------------------------------------
Expected: -1
Received: 18446744073709552000

```

It works as expected in our react-native app. 

### Can you reproduce the bug?

Always

### Version

12.1.0

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

Mac OS 13.5.1 (22G90)

### Build environment

RN 0.71.13

### Cocoapods version

1.12.1

**Comment by Yupeng-li** [comment]
Hi @kneth , it's in the latest version of realm  https://github.com/realm/realm-js/blob/main/packages/realm/src/Realm.ts#L378 and it's documented [here](https://www.mongodb.com/docs/realm-sdks/js/latest/classes/Realm-1.html#schemaVersion-2)

Thank you for pointing me to `Realm.exists`. I can call it before calling schemaVersion to bypass the issue.

**Comment by kneth** [comment]
@Yupeng-li Sorry, I was too fast to say that `schemaVersion` cannot be used. We need to write a couple of extra tests - and likely fix a bug.

You can still use `Realm.exists`.

**Comment by Yupeng-li** [comment]
@kneth No worries, thank you very much!

**Comment by kneth** [comment]
@Yupeng-li We have a bug fix in #6126
